hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -582,7 +591,7 @@ def onchange_unit_amount(self):
                 new_ts = new_ts.merge_timesheets()
             unit_amount = new_ts.unit_amount + diff_amount
             new_ts.write({'unit_amount': unit_amount})
-        else:
+        elif self._adjust_amount(diff_amount):
```

Comment:
Not sure this is correct - you're ignoring the return value because?

Reply:
If you are forbidding negative values, then you don't want to create timesheets with negative value. Thus, in this `elif` we are checking if `diff_amount` is negative or not.

Reply:
Not sure I'm following, by default `_adjust_amount` returns unchanged value, right? So, elif -1: will still execute, thus I'm not sure I follow

Reply:
Well, now this `_adjust_amount` has disappeared :)